classes/local/datasource/base.php

```diff
@@ -245,6 +245,13 @@ public function get_cached_data(): \stdClass {
         return $data;
     }
 
+    /**
+     * Get the cache used by the datasource.
+     *
+     * @return \cache
+     */
+    abstract public function get_cache(): \cache;
```

Comment:
If I'm understanding this correctly, with this implementation we cannot use the nullcache anymore? Since `get_cache()` expects a `\cache` return and isn't nullable

Reply:
I've changed this to ?\cache